File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int num, int[] lost, int[] reserve) {
        int[] clothes = new int[num+1];
        Arrays.fill(clothes,1);
        clothes[0] = 0;
        Arrays.sort(lost);
        
        for(int n : lost){
            clothes[n] -= 1 ;
        }
        
        for(int n : reserve){
            clothes[n] += 1 ;
        }
        // System.out.println(Arrays.toString(clothes));

        for(int n : lost){
            if(clothes[n] == 0){
              if(n-1 >= 0 && clothes[n-1] > 1){
                  clothes[n] += 1;
                  clothes[n-1] -= 1;
                  continue;
              }
             if(n+1 <= num && clothes[n+1] > 1){
                   clothes[n] += 1;
                   clothes[n+1] -= 1;
              }
                
            } 
        }
        
        // System.out.println(Arrays.toString(clothes));
        return (int)Arrays.stream(clothes).filter(s -> s > 0).count();
    }
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    m,k = n,n
    new_lost = set(lost) - set(reserve)
    new_reserve = set(reserve) -set(lost)
    for i in new_lost:
        if i+1 in new_reserve:
            new_reserve.remove(i+1)
        elif i-1 in new_reserve:
            new_reserve.remove(i-1)
        else:
            m -= 1
    
    new_lost = set(lost) - set(reserve)
    new_reserve = set(reserve) -set(lost)
    for i in new_lost:
        if i-1 in new_reserve:
            new_reserve.remove(i-1)
        elif i+1 in new_reserve:
            new_reserve.remove(i+1)
        else:
            k -= 1
    return max(m,k)